src/posit/connect/_api_call.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Protocol
+
+if TYPE_CHECKING:
+    from ._json import Jsonifiable
+    from .context import Context
+
+
+class ApiCallProtocol(Protocol):
+    _ctx: Context
+    _path: str
+
+    def _endpoint(self, extra_endpoint: str = "") -> str: ...
+    def _get_api(self, *, extra_endpoint: str = "") -> Jsonifiable: ...
+    def _delete_api(self, *, extra_endpoint: str = "") -> Jsonifiable | None: ...
+    def _patch_api(self, json: Jsonifiable | None, *, extra_endpoint: str = "") -> Jsonifiable: ...
+    def _put_api(self, json: Jsonifiable | None, *, extra_endpoint: str = "") -> Jsonifiable: ...
```

Comment:
I think it would be wise to lean on the Python requests Session interface. Which does runtime assertions to verify the types. Our testing strategy will reveal any issues related to the incorrect type being used.

Reply:
Moved to issue